namui/src/namui/common/types/macros.rs

```diff
@@ -76,7 +80,20 @@ macro_rules! common_for_f32_type {
         impl<T: $crate::types::Ratio> std::ops::Mul<T> for $your_type {
             type Output = $your_type;
             fn mul(self, rhs: T) -> Self::Output {
-                (self.0 * rhs.as_f32()).into()
+                (self.as_f32() * rhs.as_f32()).into()
```

Comment:
@namse We need discussion.
`Percent` act like [this](https://github.com/NamseEnt/namseent/blob/38e13c985a7d8771e9f280beb6edbe5a10674b2d/namui/src/namui/common/types/percent.rs#L4-L9). 
```rust
common_for_f32_type!(
    Percent,
    // To f32
    |lhs: Percent| -> f32 { lhs.0 / 100.0 },
    // From f32
    |lhs: f32| -> Percent { Percent(lhs * 100.0) },
    percent,
    PercentExt
);
```
It leads to situation like below.
`100 percent * 4 = 40,000 percent`, because actual behavior is
1. `(100 * 4).into()`
2. `400.into()`
3. `Percent(400 * 100)`
4. `Percent(40,000)`

Which one do you think is better?
1. Use `as_f32` while impl `Mul` (current)
5. Store without multiplying by 100 at `Percent.0`

Reply:
I'm not sure what are you talking about. Is this test cover what you concern?
![image](https://user-images.githubusercontent.com/3580430/193462954-bd1a2ff9-8a2e-4d16-a405-8654f523be79.png)

Reply:
Yes, that test shows 60,000, not 600.

Reply:
What do you mean? this test shows `600 percent` is `150 percent * 4.0`

Reply:
Test was failed before a9da757266e1ad95539ebe4e2429a854640582cd
There was problem. Percent stores 100 multiplied f32 at `Percent.0`.
But previous code used `Percent.0` directly for Mul. So test was failed with 60,000, not 600.
![image](https://user-images.githubusercontent.com/38313680/193512990-d6b7c6bd-5152-4332-9ed6-22a225da9311.png)

Reply:
All test include new test are passed well in master branch. I have no idea what you are saying.
![image](https://user-images.githubusercontent.com/3580430/193516490-3b87af1b-6c96-4d7d-aeae-37bd92a4a701.png)

Reply:
Give me a test code and code branch which fail the test

Reply:
I think it's because of `impl Ratio for f32`
https://github.com/NamseEnt/namseent/blob/a9da757266e1ad95539ebe4e2429a854640582cd/namui/src/namui/common/types/ratio.rs#L10

Reply:
Ok, `assert_eq!(600.percent(), 150.percent() * 400.percent());` is failed in master branch.

It should be change to f32 and change back again. It's not a good idea to store Percent without multiplying 100, because it should not the limit how user want to save data inside of struct.

Simply put as_f32 there please.
```rust
impl<T: $crate::types::Ratio> std::ops::Mul<T> for $your_type {
    type Output = $your_type;
    fn mul(self, rhs: T) -> Self::Output {
        (self.as_f32() * rhs.as_f32()).into()
                // ^--this
    }
}
```